### DISCUSSION TITLE
有没有示例演示同一个应用打包成不同平台的包

### DISCUSSION THREAD

**Discussion by wanghaisheng** [DioxusLabs/dioxus#444] [Q&A]
比如我开发一个切换桌面壁纸的程序，可以做成windows/linux/mac下的包，或者 web  或者打包成可以发布在android ios上的包

**Comment by rMazeiks** [comment]
Hi,

I don't speak Chinese (like many others), so here's a google-translated version of the issue:

> Is there an example of how the same app can be packaged into different platforms?
>
> For example, if I develop a program for switching desktop wallpapers, it can be made into a package under windows/linux/mac, or a web or packaged package that can be published on android ios

---

For windows/linux/mac, everything should work with the same [setup](https://dioxuslabs.com/reference/platforms/desktop).

For web, you'll need a [special setup](https://dioxuslabs.com/reference/platforms/web) that uses `trunk` to build your assets, but you should be able to reuse your existing application code. So yes, you could probably use conditional compilation to make the same app work on desktop/web (I haven't tried this myself but I don't think there are any limitations).

Dioxus doesn't have the infrastructure to run existing examples on the web, but it should be easy to make them work in a standalone project.

For Android/iOS, unfortunately, I don't think Dioxus is working at the moment.

**Comment by mrxiaozhuox** [comment]
目前我们的 Android IOS 的支持做的并不好（因为上层我们使用的框架原因），所以说我们在这里仅仅讨论 Desktop 和 Web 平台的项目。

因为 Web 使用 WASM 进行打包，所以说我们有一些 API 在 Web 平台上是有局限性的，所以说正常情况下，你很难做到 Desktop & Web 平台的同时兼容（当然通过特殊手段也是可以实现的，这会非常复杂）

## Web 打包

使用 [dioxus-cli](https://github.com/DioxusLabs/cli) 中的 build 方法就可以将你的 Web 应用打包成一个可以随时打开的网页程序。

## Desktop 打包

目前仅仅做到了可以编译成二进制应用（可以运行的那种），但如果你想打包成类似于 dmg 之类的文件，目前 Dioxus 暂时没有这方面的实现，可能需要自己通过一些手段完成。

## 题外话

我们在 Github 社区中一般会使用英文进行交流，不过如果你对 Dioxus 感兴趣的话，可以加入我创建的Q群：863409183

**Reply by onichandame** [reply]
### 题外话

是否有地方列出所有dioxus的官方讨论区？我知道的只有Github Discussion。如果有slack或gitter之类的频道或许可以给更多新手提供更及时的帮助
